File: fathersDay/math/Day2 (7. Reverse Integer)/Solution.py
class Solution:
    def Reverse(self, x: int) -> int:
        sign = -1 if x < 0 else 1
        x = abs(x)
        reversed_x = 0
        
        while x > 0:
            digit = x % 10
            reversed_x = reversed_x * 10 + digit
            x //= 10
            
        reversed_x *= sign
        if reversed_x < -2**31 or reversed_x > 2**31 - 1:
            return 0
        
        return reversed_x

solution = Solution()
print(solution.Reverse(123))        
print(solution.Reverse(-123))      
print(solution.Reverse(120))       

File: fathersDay/math/Day2 (7. Reverse Integer)/Solution.cpp
class Solution {
public:
    int reverse(int x) {
        long reversed_x = 0;
        int sign = (x < 0) ? -1 : 1;
        x = abs(x);
        
        while (x > 0) {
            int digit = x % 10;
            reversed_x = reversed_x * 10 + digit;
            x /= 10;
        }
        
        reversed_x *= sign;
        
        if (reversed_x < -2147483648 || reversed_x > 2147483647) { // -2^31 and 2^31 - 1
            return 0;
        }
        
        return (int) reversed_x;
    }
};